metadata_service/api/user.py

```diff
@@ -31,8 +36,211 @@ def __init__(self) -> None:
 
     def get(self, user_id: str) -> Iterable[Union[Mapping, int, None]]:
         try:
-            table = self.neo4j.get_user_detail(user_id=user_id)
-            return marshal(table, user_detail_fields), HTTPStatus.OK
+            user = self.neo4j.get_user_detail(user_id=user_id)
+            return marshal(user, user_detail_fields), HTTPStatus.OK
 
         except NotFoundException:
             return {'message': 'User id {} does not exist'.format(user_id)}, HTTPStatus.NOT_FOUND
+
+
+class UserBookmarkAPI(Resource):
+    """
+    Build get / put API to support user bookmark resource features.
+    It will create a relationship(follow / followed_by) between user and resources(table, dashboard etc)
+    """
+
+    def __init__(self) -> None:
+        self.neo4j = neo4j_proxy.get_neo4j()
+
+    def get(self, user_id: str) -> Tuple:
+        """
+        Return a list of resources that user has followed
+
+        :param user_id:
+        :return:
+        """
+        try:
+            resources = self.neo4j.get_resources_by_user_relation(user_id=user_id,
+                                                                  relation='FOLLOW')
```

Comment:
I think we are not ready for abstraction at this moment. I'd rather explicitly call `get_table_by_user_relation` until we design the abstraction especially because it's returning table detail payload. Thought?

Also, can we have relation as enum? It seems it will be increasing and good to keep it in one place as well as it can also hide information of bi-direction relation value from this method.